errors.go

```diff
@@ -0,0 +1,84 @@
+package rpc
+
+import "errors"
+
+// ResponseErr is an enum type for providing error type
+// information over the wire between rpc server and client.
+type ResponseErr int
+
+const (
+	// NonRPCErr is an error that hasn't arisen from the gorpc package.
+	NonRPCErr ResponseErr = iota
+	// ServerErr is an error that has arisen on the server side.
+	ServerErr
+	// ClientErr is an error that has arisen on the client side.
+	ClientErr
+)
+
+// ServerError indicates that error originated in server
+// specific code.
+type ServerError struct {
```

Comment:
I think we don't need to have a few of these as exported types/methods. Just having `IsServer(error) bool`, `IsClient(error) bool` and `IsRPC(error) bool` should be all the API that we need to expose? All the rest (error types etc) seem to me internals of this library... if we need a serializable exported type in `Response` for `ResponseErr`, we can just use an `int` instead.

Reply:
Done.